packages/xo-web/src/xo-app/vm/tab-disks.js

```diff
@@ -103,6 +106,7 @@ const COLUMNS_VM_PV = [
       return (
         sr !== undefined && (
           <XoSelect
+            compareOptions={compareSr}
```

Comment:
Don't you need to sort the containers here instead?

Reply:
here we have only SR of the pool and this to have shared SR first in selector.

Reply:
Yes, but they're still grouped by containers so if the containers aren't sorted (e.g. the pool comes last), then the shared SRs (whose container is the pool) won't come first in the list.

Reply:
[We concat pools to hosts  so we have pool first](https://github.com/vatesfr/xen-orchestra/blob/master/packages/xo-web/src/common/select-objects.js#L440ToL444)

Reply:
Then either we trust `SelectSr` for sorting the containers with the pools first (which I don't think is a good idea because it's not *supposed* to), either we don't trust it and we sort the containers ourselves here. But in both cases, there's no need to sort the SRs because all the SRs in a pool are shared and all the SRs in a host are *not* shared.

Reply:
(For SR in pool, it may be an SR that doesn't have PBDs)[https://github.com/vatesfr/xen-orchestra/blob/master/packages/xo-server/src/xapi-object-to-xo.js#L461]

Reply:
So we don't trust and we add compareContainers and compareOptions.

Reply:
Ok, I didn't remember the `!obj.$PBDs[0]` condition and I don't really know what it means for a SR to both not be shared and have 0 PBDs :thinking: 

So yes, let's keep the `compareOptions` just to be sure and add `compareContainers` because `SelectSr` is not supposed to know how we want to sort them.